### PR TITLE
Updated live thumbnail code to work with LiveNG

### PR DESCRIPTION
**Pull request by nadav-ravid**
LIV-181 #time 2d

### CODE REVIEW

File: infra/media/KThumbnailCapture.php [hunk 1]
```diff
@@ -18,9 +18,10 @@ public static function shouldResizeByPackager($params, $type, $dimension)
 	}
 
 
-	public static function generateThumbUrlWithOffset($url, $calc_vid_sec, $packagerCaptureUrl, $capturedThumbPath, $width = null, $height = null, $offsetPrefix = '', $postFix = '')
+	public static function generateThumbUrlWithOffset($url, $calc_vid_sec, $packagerCaptureUrl, $capturedThumbPath, $width = null, $height = null, $offsetPrefix = '', $postFix = '', $offsetPostfix='')
 	{
 		$offset = floor($calc_vid_sec * 1000);
+		$offset .= $offsetPostfix;
```
Comment (david-winder-kaltura) [inline infra/media/KThumbnailCapture.php:24]: let concat to `$offset = $offsetPrefix . $offset . $offsetPostfix;`
And separate to small function for build the string

File: alpha/apps/kaltura/lib/myPackagerUtils.class.php [hunk 2]
```diff
@@ -333,28 +335,60 @@ protected static function captureMappedThumb($entry, $flavorAsset, $capturedThum
...
+		foreach ($currentEntryServerNodes as $entryServerNode)
+		{
+			$serverNode = ServerNodePeer::retrieveActiveMediaServerNode(null, $entryServerNode->getServerNodeId());
+			if (!$serverNode) {
+				continue;
+			}
+
+			$serverNodeUrl = str_replace('{dc}', $serverNode->getEnvDc(), $liveCaptureUrl);
+
+			$tokenGenerator = new DeliveryProfileLivePackagerHls();
```
Comment (david-winder-kaltura) [inline alpha/apps/kaltura/lib/myPackagerUtils.class.php:369]: separate to small func that return the token string

File: alpha/apps/kaltura/lib/myPackagerUtils.class.php [hunk 3]
```diff
@@ -333,28 +335,60 @@ protected static function captureMappedThumb($entry, $flavorAsset, $capturedThum
 	 * @return bool|mixed
 	 * @throws Exception
 	 */
-	protected static function captureLiveThumb(entry $entry, $liveType, $destThumbPath, $calc_vid_sec, $width = null, $height = null)
+	protected static function captureLiveThumb(entry $entry, $destThumbPath, $calc_vid_sec, $width = null, $height = null)
 	{
-		$packagerCaptureUrl = self::getPackagerUrlFromConf(kPackagerUrlType::LOCAL_LIVE_THUMB);
-		if (!$packagerCaptureUrl)
+		$result = false;
```
Comment (david-winder-kaltura) [inline alpha/apps/kaltura/lib/myPackagerUtils.class.php:340]: as we don't assign result a value let not define it but just return the bool value when needed

File: alpha/apps/kaltura/lib/myPackagerUtils.class.php [hunk 4]
```diff
@@ -333,28 +335,60 @@ protected static function captureMappedThumb($entry, $flavorAsset, $capturedThum
...
-		$dc = myEntryUtils::getLiveEntryDcId($entry->getRootEntryId(), EntryServerNodeType::LIVE_PRIMARY);
-		if (is_null($dc))
+		$liveCaptureUrl = self::getPackagerUrlFromConf(kPackagerUrlType::LIVE_THUMB);
+		if (!$liveCaptureUrl)
 		{
-			return false;
+			return $result;
 		}
+		$timelineId = $entry->getId();
+		$liveCaptureUrl .= "e/{$entry->getRootEntryId()}/tl/$timelineId/";
```
Comment (david-winder-kaltura) [inline alpha/apps/kaltura/lib/myPackagerUtils.class.php:352]: lets move it to the foreach and then under the serverNode (liveCluster)

File: alpha/apps/kaltura/lib/myPackagerUtils.class.php [hunk 5]
```diff
@@ -333,28 +335,60 @@ protected static function captureMappedThumb($entry, $flavorAsset, $capturedThum
 	 * @return bool|mixed
 	 * @throws Exception
 	 */
-	protected static function captureLiveThumb(entry $entry, $liveType, $destThumbPath, $calc_vid_sec, $width = null, $height = null)
+	protected static function captureLiveThumb(entry $entry, $destThumbPath, $calc_vid_sec, $width = null, $height = null)
```
[the remark this answers is not in the document]
Comment (nadav-ravid) [reply alpha/apps/kaltura/lib/myPackagerUtils.class.php:338]: yes the function is called exclusively for live related entries

**Pull request merged**